### PR TITLE
Added a much simpler version of connrequester for developers machines

### PR DESCRIPTION
**Pull request by clare**
This is to get around having to have modemd running on machines which it won't work on anyway. 

Also updated the readme.

### CODE REVIEW

File: connrequester/connrequester_linux_amd64.go [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+// if no connection is made.
+// timeout is the time given to make a connection each try.
+// retryAfter is the duration between attempts, it will multipy by 1.5 after
+// each try with a maximum of 2 hours.
+// maxRetries is how many times it will try to make a connection. If -1 it will
+// try until a connection is made.
+func (cr *ConnectionRequester) WaitUntilUpLoop(
+	timeout time.Duration,
+	retryAfter time.Duration,
+	maxRetries int) error {
```
Comment (mjs) [inline connrequester/connrequester_linux_amd64.go]: You can leave out all the parameter names if you're not using any of them

**Review by mjs — APPROVED** [review]
LGTM.

Please confirm that it's still working as before on the pi before merging :)

**Review by CameronRP — APPROVED** [review]
Looks good. You don't have all of the public methods for ConnectionRequester but I am guessing this is not an issue for your testing?

File: connrequester/connrequester_nomodemd.go [hunk 2]
```diff
@@ -0,0 +1,38 @@
+// +build linux/amd64
```
Comment (mjs) [inline connrequester/connrequester_nomodemd.go]: This isn't correct (and is why the build is failing in Travis). Platforms and architectures can't be combined with a slash.

Perhaps just go with `+build amd64` and `+build !amd64` for the other file.

Comment (mjs) [reply connrequester/connrequester_nomodemd.go]: The `_nomodemd.go` suffix is great. Much clearer.

**Comment by clare** [comment]
I have confirmed now that this is compiling for all projects that use it and it works on the pi with thermal-uploader.

**Pull request merged**